dipy/reconst/mdki.py

```diff
@@ -0,0 +1,426 @@
+        in its last axis
+    gtab : a GradientTable class instance
+        The gradient table for this prediction
+    S0 : float or ndarray (optional)
+        The non diffusion-weighted signal in every voxel, or across all
+        voxels. Default: 1
+
+    Notes
+    -----
+    The predicted signal is given by:
+        $MS(b) = S_0 * exp(-bD + 1/6 b^{2} D^{2} K)$, where $D$ and $K$ are the
+        mean signal diffusivity and mean signal kurtosis.
+
+    References
+    ----------
+    .. [1] Henriques, R.N., 2018. Advanced Methods for Diffusion MRI Data
+           Analysis and their Application to the Healthy Ageing Brain (Doctoral
+           thesis). Downing College, University of Cambridge.
+           https://doi.org/10.17863/CAM.29356
+    """
+    # Define MDKI design matrix given gtab.bvals
+    A = design_matrix(round_bvals(gtab.bvals))
+
+    # Initialize pred_sig
+    pred_sig = np.zeros(mdki_params.shape[:-1] + gtab.bvals.shape)
+
+    # looping for all voxels
+    index = ndindex(mdki_params.shape[:-1])
+    for v in index:
+        params = mdki_params[v].copy()
+        params[1] = params[1] * params[0] ** 2
+        if isinstance(S0, np.ndarray):
+            S0v = S0[v]
+        else:
+            S0v = S0
+        pred_sig[v] = S0v * np.exp(np.dot(A[:, :2], params))
+
+    return pred_sig
+
+
+class MeanDiffusionKurtosisModel(ReconstModel):
+    """ Mean signal Diffusion Kurtosis Model
+    """
+
+    def __init__(self, gtab, bmag=None, return_S0_hat=False, *args, **kwargs):
+        """ Mean Signal Diffusion Kurtosis Model [1]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+
+        bmag : int
+            The order of magnitude that the bvalues have to differ to be
+            considered an unique b-value. Default: derive this value from the
+            maximal b-value provided: $bmag=log_{10}(max(bvals)) - 1$.
+
+        return_S0_hat : bool
+            If True, also return S0 values for the fit.
+
+        args, kwargs : arguments and keyword arguments passed to the
+        fit_method. See mdti.wls_fit_mdki for details
+
+        min_signal : float
+            The minimum signal value. Needs to be a strictly positive
+            number. Default: 0.0001.
+
+        References
+        ----------
+        .. [1] Henriques, R.N., 2018. Advanced Methods for Diffusion MRI Data
+               Analysis and their Application to the Healthy Ageing Brain
+               (Doctoral thesis). Downing College, University of Cambridge.
+               https://doi.org/10.17863/CAM.29356
+        """
+        ReconstModel.__init__(self, gtab)
+
+        self.return_S0_hat = return_S0_hat
+        self.ubvals = unique_bvals(gtab.bvals, bmag=bmag)
+        self.design_matrix = design_matrix(self.ubvals)
+        self.bmag = bmag
+        self.args = args
+        self.kwargs = kwargs
+        self.min_signal = self.kwargs.pop('min_signal', MIN_POSITIVE_SIGNAL)
```

Comment:
ok, I see, I wonder whether it will be better to be explicit and just add this parameter in the  `__init__` function like you did [on line 329](https://github.com/nipy/dipy/pull/1230/files#diff-0e7e49b7117951792bdf687bed1a384eR329)